### ISSUE TITLE
Stricter checks for ECS task definitions

### ISSUE THREAD

**Issue by acdha** [aws-cloudformation/cfn-python-lint#855]
*cfn-lint version: (`cfn-lint 0.19.1`)*

The CF spec files defines `AWS::ECS::TaskDefinition` as a string:

```json
   "AWS::ECS::TaskDefinition": {
"ExecutionRoleArn": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-taskdefinition.html#cfn-ecs-taskdefinition-executionrolearn",
          "PrimitiveType": "String",
          "Required": false,
          "UpdateType": "Immutable"
        },
```

Unfortunately, the actual service expects that to be a full ARN, as the name implies, so it'll take a few minutes before you'll realize your mistake if you just use the short name as used in the UI. It'd be really nice if `cfn-lint` could catch this kind of error. On a related note touching on #238, this also made me think it'd be really helpful if there was a way to require this to be a reference or substitution to catch copy-and-paste errors.

**Comment by kddejong** [comment]
This should be easy with our current definitions.  We have some issues with Role ARNs also taking the name.  In your scenario did the Role have a Path that went beyond `/`.

**Comment by acdha** [comment]
This is what I ended up having in my template:

```yaml
"ExecutionRoleArn": {
    "Fn::Sub": "arn:aws:iam::${AWS::AccountId}:role/SentryECSTaskExecutionRole"
},
```
